config/routes.rb

```diff
@@ -32,11 +32,14 @@
     put 'dashboard_show_recent_activity', to: 'dashboard#show_recent_activity'
 
 
-    get 'app_configuration', to: 'app_configuration#show'
-    put 'app_configuration', to: 'app_configuration#update'
+    get 'app_configuration/(:id)', to: 'app_configuration#show',
+        as: :app_configuration
 
-    get 'app_configuration/redigera', to: 'app_configuration#edit',
+    get 'app_configuration/:id/redigera', to: 'app_configuration#edit',
         as: :edit_app_configuration
+    put 'app_configuration/:id', to: 'app_configuration#update',
+        as: :update_app_configuration
```

Comment:
Why add `update` to the path name?  Designating just the resource, and the operation, is RESTful.

Reply:
@patmbolger  The current paths for `app_configuration` are non-standard.  Specifically, both of these lines are non-standard:
```
put 'app_configuration', to: 'app_configuration#update'
get 'app_configuration/redigera', to: 'app_configuration#edit', as: :edit_app_configuration
```
So to continue working with the routes specified as those 2 lines, I needed to use another non-standard route.

I would much rather stick with the Rails convention/standard routing and instead use the following:
```
  # We don't have a view for :index or :destroy.  Currently, admins only
  # view and edit the current AppConfiguration.
  resources :app_configuration, only: [:show, :edit, :update]
```
(which means changing the routes used in menu navigation, which I think is fine)

Reply:
Those proposed routes are member routes for a collection - AppConfig is not a collection.  There should not be an `:id` parameter in the routes.

Reply:
I'm feeling frustrated by this.  Can we schedule a time to chat in Slack?  

This is a non-conventional approach: to limit an AR to a single _instance_ but not use either class methods or Singletons.  Without any documentation about it, I went with convention.   
We chatted briefly about this during the Monday dev meeting, but clearly we are not communicating about it well.

Either we need to clearly document the business _and_ development case for using a singleton here (e.g. only 1 instance ever with no historical record of any kind) or go with convention:  e.g. it can have more than 1 instance recorded in the AR table, which IMO fulfills the need to show simple historical record, which is key when dealing something so critical as changing the configuration of the entire application.
  (By "development case" I mean that the business case trumps any cost in development knowledge required and maintenance complexity.)

I'll also post a pointer to this in Slack  and hopefully we can work out a time to chat.

Reply:
agreed on:

**AppConfiguration should be a Singleton.**
There should only ever be 1 entry (row/record) in the db for it.
Thus no `:id` should be needed for any of the routes.

(I'll get back to work on this now that this is clarified.)